private/pkg/manifest/module.go

```diff
@@ -24,6 +28,50 @@ var hashKindToDigestType = map[modulev1alpha1.HashKind]DigestType{
 	modulev1alpha1.HashKind_HASH_KIND_SHAKE256: DigestTypeShake256,
 }
 
+type Blob interface {
+	Digest() *Digest
+	Open(ctx context.Context) (io.ReadCloser, error)
+}
+
+type memoryBlob struct {
+	digest  Digest
+	content []byte
+}
+
+var _ Blob = (*memoryBlob)(nil)
+
+// NewMemoryBlob takes a digest and a content, and turns it into an in-memory
+// representation of a blob, which returns the digest and an io.ReadCloser for
+// its content. It optionally validates that the passed digest and content
+// match.
+func NewMemoryBlob(digest Digest, content []byte, validateHash bool) (Blob, error) {
+	if validateHash {
```

Comment:
I absolutely agree that blobs should have the option to validate. This is fine but I suggest a functional option pattern for validateHash. Reason being is validating from a stream can have one of two ways of validating. If it's seekable, it can be validated on construction. If not, it can only be validated with an error code on `Close`. So we have two ways of specifying validation: on construction and on close. Having a functional option for validation would make for a touch of consistency.[1]

Minor though. We can do this when we get to saving/reading blobs from persistent storage (cache, in the BSR, etc).

[1] Even when you have seekable input, validating on close can have performance impact, even with memory! Scanning memory twice will blow the CPU cache.

Reply:
Done, ptal